### PR TITLE
Everywhere: Convert more stuff to `LibFileSystem`/`Core::File`

### PR DESCRIPTION
**Pull request by cammo1123**
Part (2/n)

Trying to do this in chunks, so it's easier to review. If these need to have more changes or broader commits, let me know.

Found an issue in Piano where the exported WAV is corrupted on master. However, my changes have not affected the issue (I will make an issue if needed).

### CODE REVIEW

**Comment by LucasChollet** [comment]
> Found an issue in Piano where the exported WAV is corrupted on master. However, my changes have not affected the issue (I will make an issue if needed).

How can you be sure of that? These files are eventually corrupted twice and there are no way to test. I would be in favor to drop the commit waiting for the initial bug to be fixed.

**Comment by cammo1123** [comment]
> These files are eventually corrupted twice and there are no way to test. I would be in favor to drop the commit waiting for the initial bug to be fixed.

A G in Piano was created in both master and this pr. They output the same file size. However, they both crash the internal audio player and arent the right note.

<details><summary>Details</summary>
<p>
<h1>TURN DOWN VOLUME, EXTREMLY LOUD</h1>

Command used to convert: `ffmpeg -i ThisPR.wav ThisPR.mp4`

<a href="https://user-images.githubusercontent.com/36214028/227574527-567f0275-6f5b-49e7-afb0-8e3ce5a2d178.mp4">
Master.mp4</a>
<a href="https://user-images.githubusercontent.com/36214028/227575531-ee36eaa3-4ac6-490b-8213-b2185fd790a3.mp4">
This PR.mp4</a>
</p>
</details>

**Comment by LucasChollet** [comment]
It looks to me a bit like a garbage in garbage out situation, that's only my opinion though

**Comment by cammo1123** [comment]
> It looks to me a bit like a garbage in garbage out situation, that's only my opinion though

Should have made the examples a bit more interesting, if you put different notes it you can hear it changing notes. But I'll probably end up removing that commit. Maybe wait for someone else to fix it. Audio stuff is way way way over my head

**Review by caoimhebyrne — APPROVED** [review]
Looks good to me from a general glance.

Unsure of the massive changes in [Userland/Libraries/LibCpp/Tests/parser/strace.ast](https://github.com/SerenityOS/serenity/pull/18026/files#diff-435c293d4ee5e163d68b51d074d88ef10d98e1d2a5ed7eacc8670148b9e177f3) however. An explanation would be appreciated (unless this is normal when editing the strace tests, and if so, everything else seems fine to me)

**Comment by linusg** [comment]
@caoimhebyrne that file contains a text dump of the AST of the `strace.cpp` utility, generated as a test case by/for our `LibCpp`. Changes to `Userland/Libraries/LibCpp/Tests/parser/strace.cpp` require updating the expected AST, although I doubt we actually build that file. Regardless, converting test files away from deprecated APIs is nice :)

**Comment by linusg** [comment]
(it might have been a copy of the actual `strace` - https://github.com/SerenityOS/serenity/blob/master/Userland/Utilities/strace.cpp - once, but definitely not anymore)

**Comment by caoimhebyrne** [comment]
That makes sense now, thank you for the explanation @linusg :) sorry for adding more noise here :P

File: Userland/Libraries/LibAudio/WavWriter.h [hunk 1]
```diff
@@ -11,42 +11,38 @@
...
 
-class WavWriter {
+class WavWriter final : public Core::Object {
     AK_MAKE_NONCOPYABLE(WavWriter);
     AK_MAKE_NONMOVABLE(WavWriter);
+    C_OBJECT(WavWriter);
 
 public:
-    WavWriter(StringView path, int sample_rate = 44100, u16 num_channels = 2, u16 bits_per_sample = 16);
+    static ErrorOr<NonnullRefPtr<WavWriter>> try_create_from_file(StringView path, int sample_rate = 44100, u16 num_channels = 2, u16 bits_per_sample = 16);
```
Comment (linusg) [inline Userland/Libraries/LibAudio/WavWriter.h:26]: `try_` is reserved for APIs that already have an infallible variant and need to be distinguished by name, _not_ as a general prefix for fallible functions

File: Userland/Libraries/LibAudio/WavWriter.h [hunk 2]
```diff
@@ -11,42 +11,38 @@
...
 #include <LibAudio/Sample.h>
-#include <LibCore/DeprecatedFile.h>
+#include <LibCore/File.h>
 #include <LibCore/Forward.h>
+#include <LibCore/Object.h>
 
 namespace Audio {
 
-class WavWriter {
+class WavWriter final : public Core::Object {
```
Comment (linusg) [inline Userland/Libraries/LibAudio/WavWriter.h:20]: And this is needed why...?

**Comment by kleinesfilmroellchen** [comment]
> Found an issue in Piano where the exported WAV is corrupted on master. However, my changes have not affected the issue (I will make an issue if needed).

Known issue; fixed in #17768 .

**Comment by cammo1123** [comment]
> > Found an issue in Piano where the exported WAV is corrupted on master. However, my changes have not affected the issue (I will make an issue if needed).
> 
> Known issue; fixed in #17768 .

Should I rebase on top of that PR? And wait for that to be merged? Or should I leave as is?

**Comment by kleinesfilmroellchen** [comment]
> > > Found an issue in Piano where the exported WAV is corrupted on master. However, my changes have not affected the issue (I will make an issue if needed).
> > 
> > Known issue; fixed in #17768 .
> 
> Should I rebase on top of that PR? And wait for that to be merged? Or should I leave as is?

Ignore the issue; there are no fundamental conflicts.

**Review by kleinesfilmroellchen — APPROVED** [review]

**Review by ADKaster — APPROVED** [review]

**Pull request merged**